testing/e2e/agent_install_test.go

```diff
@@ -0,0 +1,348 @@
+// Additionally the MacOS specific elastic-agent symlink will be recreated with a hard link.
+func (suite *AgentInstallSuite) extractLinux(r io.Reader) {
+	suite.T().Helper()
+	fleetPath := ""
+	// Extract tar.gz stream
+	stream, err := gzip.NewReader(r)
+	suite.Require().NoError(err)
+	tarReader := tar.NewReader(stream)
+	for header, err := tarReader.Next(); err == nil; header, err = tarReader.Next() {
+		if header.FileInfo().IsDir() {
+			err := os.Mkdir(filepath.Join(suite.downloadPath, header.Name), 0755)
+			suite.Require().NoError(err)
+		} else {
+			if !header.FileInfo().Mode().IsRegular() {
+				// Linux archives should not have symlinks
+				suite.T().Logf("unable to extract %s", header.Name)
+				continue
+			}
+			dst, err := os.Create(filepath.Join(suite.downloadPath, header.Name))
+			suite.Require().NoError(err)
+			err = dst.Chmod(header.FileInfo().Mode())
+			suite.Require().NoError(err)
+			_, err = io.Copy(dst, tarReader)
+			dst.Close() // might be a dirty close
+			suite.Require().NoError(err)
+
+			if strings.HasSuffix(header.Name, binaryName) {
+				fleetPath = filepath.Join(suite.downloadPath, header.Name)
+			}
+		}
+	}
+	// Copy fleet-server binary to un archived package
+	suite.Require().NotEmpty(fleetPath, "no fleet-server component detected")
+	err = os.Remove(fleetPath)
+	suite.Require().NoError(err)
+	err = os.Link(suite.binaryPath, fleetPath)
+	suite.Require().NoError(err)
+}
+
+func (suite *AgentInstallSuite) TearDownSuite() {
+	if suite.downloadPath != "" {
+		err := os.RemoveAll(suite.downloadPath)
+		suite.Require().NoErrorf(err, "failed to remove download from %s", suite.downloadPath)
+	}
+}
+
+func (suite *AgentInstallSuite) SetupTest() {
+	if suite.installDetected {
+		suite.T().Skip("elastic-agent install detected, skipping test.")
+	}
+
+	portFree := suite.IsFleetServerPortFree()
+	suite.Require().True(portFree, "port 8220 must not be in use for test to start")
+}
+
+func (suite *AgentInstallSuite) TearDownTest() {
+	if suite.T().Skipped() {
+		return
+	}
+	path, err := exec.LookPath(suite.agentName)
+	if err != nil {
+		suite.T().Logf("unable to detect elastic-agent install on test tear-down: %s", err)
+		return
+	}
+	err = exec.Command(path, "uninstall", "--force").Run()
+	suite.Require().NoError(err, "elastic-agent uninstall failed.")
+}
+
+func (suite *AgentInstallSuite) TestHTTP() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*3)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, filepath.Join(suite.downloadPath, suite.agentName), "install",
+		"--fleet-server-es="+suite.esHosts,
+		"--fleet-server-service-token="+suite.serviceToken,
+		"--fleet-server-insecure-http=true",
+		"--fleet-server-host=0.0.0.0",
+		"--fleet-server-policy=fleet-server-policy",
+		"--non-interactive")
+	cmd.Env = []string{"GOCOVERDIR=" + suite.coverPath} // TODO Check if this env var will be passed by the agent to fleet-server
```

Comment:
I don't think that this env var gets passed from the `install` command to the agent that will be running, or the fleet-server that is started from it